include/zephyr/bluetooth/audio/bap.h

```diff
@@ -1706,6 +1726,9 @@ struct bt_bap_scan_delegator_add_src_param {
 	/** The periodic adverting sync */
 	struct bt_le_per_adv_sync *pa_sync;
 
+	/** The broadcast isochronous group */
+	struct bt_iso_big *big;
+
```

Comment:
No, I think it may be better to find a way to lookup in the BIG from existing broadcast sink objects. 

If there aren't already a function to do so, then perhaps that needs to be implemented (as an internal function) in `bap_broadcast_sink.c` so that the scan delegator can retrieve it via that. 

The reason for this is that the `big` isn't expose to the application via the Broadcast Sink API, so adding it to the public API here will be effectively useless. 

Furthermore, there are some weird requirements in BAP that the BAP Broadcast Sink requires the BAP Scan Delegator, but you can actually have a BAP Scan Delegator without the BAP Broadcast Sink. In this case, I'm unsure how to handle it.

Reply:
@Thalley The scan delegator doesn't know anything about the broadcast sink though in the application as it's higher in the stack. I'm not sure if it should. I mean include broadcast sink headers in the scan delegator and directly call broadcast sink functions from the scan delegator?

Would an option be to register a new callback from the sink to the delegator, and the delegator can possibly discover the BIG this way?

Reply:
> I mean include broadcast sink headers in the scan delegator and directly call broadcast sink functions from the scan delegator?

Yes, we can implement such a function in `bap_broadcast_sink.c` that is only internally available. 

> Would an option be to register a new callback from the sink to the delegator, and the delegator can possibly discover the BIG this way?

That is also possible - Either way really works. 

I was looking at the implementations, and I think the easiest way to achieve this would be to use the `broadcast_sink_get_by_pa` function from the scan delegator, since the scan delegator already has the pointer to the PA sync object. 

So it would be something like
```c
struct bt_bap_broadcast_sink *broadcast_sink = broadcast_sink_get_by_pa(internal_state->pa_sync);

if (broadcast_sink != NULL && broadcast_sink->big != NULL) {
    /* BIG is being synced */
}
```

Reply:
@Thalley broadcast_sink_get_by_pa currently is an internal function part of the Broadcast Sink module. Do you mean to make it public and have the Scan Delegator call it? The Scan Delegator module would then have a dependency on the Broadcast Sink module.

Reply:
> Do you mean to make it public and have the Scan Delegator call it?

I would make it internally available in the subsystem, yes. 

> The Scan Delegator module would then have a dependency on the Broadcast Sink module. Currently it has none.

True, but you can guard the entire check with `IS_ENABLED(CONFIG_BT_BAP_BROADCAST_SINK)` when performing the checks for active BIGs. It doesn't really make sense to worry about the BIGs if we don't have the broadcast sink. 
If we wanted an alternative approach, we should implement the same functionality (although in a different way) at the ISO layer -> Basically implement a `bt_iso_foreach_big` function to iterate on all the BIGs and a `bt_iso_big_get_info` function to get the state an PA sync object of a BIG, and then perform the check based on that.